stellarphot/photometry/profiles.py

```diff
@@ -0,0 +1,305 @@
+
+    max_iters : int, optional
+        Maximum number of iterations to go through in finding the center.
+
+    match_limit : int, optional
+        Maximum number of pixels to allow the COM centroid and Gaussian center to
+        differ.
+
+    Returns
+    -------
+
+    cen : array
+        The position of the star, in pixels, as found by the centroiding
+        algorithm.
+    """
+    pad = cutout_size // 2
+    x, y = center_guess
+
+    # Keep track of iterations
+    cnt = 0
+
+    # Grab the cutout...
+    sub_data = Cutout2D(image, center_guess, (cutout_size, cutout_size), mode="trim")
+    # ...do stats on it...
+    _, sub_med, _ = sigma_clipped_stats(sub_data.data)
+    # ...and centroid.
+    mask = (sub_data.data - sub_med) < 0
+    x_cm, y_cm = centroid_com(sub_data.data - sub_med, mask=mask)
+
+    # Translate centroid back to original image (maybe use Cutout2D instead)
+    cen = np.array(sub_data.to_original_position((x_cm, y_cm)))
+    print(f"{cen=} {x_cm=} {y_cm=}")
+    # ceno is the "original" center guess, set it to something nonsensical here
+    ceno = np.array([-100, -100])
+
+    while cnt <= max_iters and (
+        np.abs(np.array([x_cm, y_cm]) - pad).max() > 3 or np.abs(cen - ceno).max() > 0.1
+    ):
+        try:
+            sub_data = Cutout2D(image, cen, (cutout_size, cutout_size), mode="trim")
+        except NoOverlapError:
+            raise RuntimeError(
+                f"Centroid finding failed, previous was {ceno}, current is {cen}"
+            )
+        _, sub_med, _ = sigma_clipped_stats(sub_data.data)
+
+        mask = (sub_data.data - sub_med) < 0
+        x_cm, y_cm = centroid_com(sub_data.data - sub_med, mask=mask)
+        ceno = cen
+        cen = np.array(sub_data.to_original_position((x_cm, y_cm)))
+        if not np.all(~np.isnan(cen)):
+            raise RuntimeError(
+                f"Centroid finding failed, previous was {ceno}, current is {cen}"
+            )
+        cnt += 1
+
+    # Is we hit the max number of iterations, raise an error
+    if max_iters > 1 and cnt > max_iters:
+        raise RuntimeError(f"Centroid finding did not converge")
+
+    # Get the final centroid position by fitting a gaussian
+    # We may not have converged on a star, so capture any warning about a
+    # bad fit.
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        ceng = centroid_2dg(sub_data.data - sub_med)
+    ceng = np.array(sub_data.to_original_position(ceng))
+
+    # Confirm that we actually found a star by comparing two centroiding
+    # methods
+    if np.linalg.norm(cen - ceng) > match_limit:
+        raise RuntimeError(
+            "Centroid did not converge on a star. "
+            f"Got {cen} from centroid_com and {ceng} from centroid_2dg."
+        )
+
+    return ceng
+
+
+class CenterAndProfile:
+    """
+    Class to dentermine center of and hold radial profile information for a star.
```

Comment:
unless you are planning to really make code that bites, you want to 'determine center' not 'dentermine center' :)